packages/host/app/services/realm-server.ts

```diff
@@ -173,9 +176,72 @@ export default class RealmServerService extends Service {
       });
   }
 
-  async fetchCreditInfo(): Promise<CreditInfo> {
+  async fetchSubscriptionData() {
+    if (this.subscriptionData) {
+      return;
+    }
+    await this.fetchSubscriptionDataTask.perform();
+  }
+
+  async handleRealmServerEvent(maybeRealmServerEvent: Partial<IEvent>) {
+    let sessionRoomId = await this.getSessionRoomId();
+    if (
+      maybeRealmServerEvent.room_id !== sessionRoomId ||
+      maybeRealmServerEvent.type !== 'm.room.message' ||
+      maybeRealmServerEvent.content?.msgtype !==
+        'org.boxel.realm-server-event' ||
+      maybeRealmServerEvent.content?.body !== 'billing-notification'
+    ) {
+      return;
+    }
```

Comment:
Also, re-order the conditional checks here so that we don't get the room session ID unless we actually have something to process. This should make it so that most tests don't need the extra setupRealmServerEndpoints hook.